src/com/untamedears/PrisonPearl/managers/WorldBorderManager.java

```diff
@@ -0,0 +1,152 @@
+    	return !border.insideBorder(loc);
+    }
+    
+    public boolean isOnWhiteList(Location loc){
+    	for(Location whLoc : WhiteListedLocations){
+    		if(whLoc.equals(loc)){
+    			return true;
+    		}
+    	}
+    	return false;
+    }
+    
+    public boolean isDirty() {
+		return dirty;
+	}
+    
+	@Override
+	public void save(File file) throws IOException {
+		FileOutputStream  fos = new FileOutputStream (file);
+		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos));
+		for(Location loc : WhiteListedLocations){
+			bw.append(loc.getX()+","+loc.getY()+","+loc.getZ());
+			bw.append("\n");
+		}
+		bw.flush();
+		bw.close();	
+		dirty = false;
+	}
+	@Override
+	public void load(File file) throws IOException {
+		FileInputStream fis = new FileInputStream(file);
+		BufferedReader br = new BufferedReader(new InputStreamReader(fis));
+		if (file.exists()){
+			if(br.readLine() == null){
+				plugin.getLogger().log(Level.INFO, file.getName()+ " file is empty");;
+				br.close();
+				return;
+			}
+			String line;
+			while ((line = br.readLine()) != null) {
+				plugin.getLogger().log(Level.INFO, ("Reading curLine: " + line));
+				String parts[] = line.split(",");
+				if(parts.length==3){
+					try{
+						Double x = stringToDouble(parts[0]);
+						Double y = stringToDouble(parts[1]);
+						Double z = stringToDouble(parts[2]);
+						Location loc = new Location(plugin.getFreeWorld(), x, y, z);
+						WhiteListedLocations.add(loc);
+						plugin.getLogger().log(Level.INFO, ("Adding location=" +loc));
+					}catch(NumberFormatException e){
+						plugin.getLogger().log(Level.WARNING, "Current line is not in the right format");
+						e.printStackTrace();
+					}
+				}else{
+					plugin.getLogger().log(Level.WARNING, "Current line is not in the right format");
+				}
+			}
+			br.close();
+			fis.close();
+		}
+		dirty = false;
+	}
+	
+    private Double stringToDouble(String str) throws NumberFormatException{
+    	return Double.parseDouble(str);
+    }
+    
+    public boolean addWhitelistedLocation(Location loc){
+    	WhiteListedLocations.add(loc);
+    	dirty=true;
+    	return true;
+    }
+    public boolean removeWhitelistedLocation(Location loc){
+    	for(Location whLoc : WhiteListedLocations){
+    		if(whLoc.equals(loc)){
+    			WhiteListedLocations.remove(whLoc);
+    			dirty=true;
+    			return true;
+    		}
+    	}
+    	return false;
+    }
+    public boolean addWhitelistedLocation(String x, String y, String z){
+    	try{
+    		Double xd = stringToDouble(x);
+			Double yd = stringToDouble(y);
+			Double zd = stringToDouble(z);
+    		WhiteListedLocations.add(new Location(plugin.getFreeWorld(), xd , yd, zd));
+        	dirty=true;
+        	return true;
+    	}catch(NumberFormatException e){
+    		return false;
+    	}
+
+    }
+    
+    public boolean removeWhitelistedLocation(String x, String y, String z){
+    	try{
+    		Double xd = stringToDouble(x);
+			Double yd = stringToDouble(y);
```

Comment:
thanks Dan, just resubmit and let me know your you are ready Black

On Tue, May 26, 2015 at 10:57 AM Daniel Boston <notifications@github.com>
wrote:

> In src/com/untamedears/PrisonPearl/managers/WorldBorderManager.java
> <https://github.com/Civcraft/PrisonPearl/pull/47#discussion_r31049577>:
>
> > +    		Double xd = stringToDouble(x);
> > +			Double yd = stringToDouble(y);
> > +			Double zd = stringToDouble(z);
> > +    		WhiteListedLocations.add(new Location(plugin.getFreeWorld(), xd , yd, zd));
> > +        	dirty=true;
> > +        	return true;
> > +    	}catch(NumberFormatException e){
> > +    		return false;
> > +    	}
> > +
> > +    }
> > +
> > +    public boolean removeWhitelistedLocation(String x, String y, String z){
> > +    	try{
> > +    		Double xd = stringToDouble(x);
> > +			Double yd = stringToDouble(y);
>
> That is odd! I've been struggling with the same on some of the work in the
> snitch visualizer, I use tabs but Scuwr appears to have used spaces. I've
> got a bunch of cleanup to do there, but in Eclipse is just looks "the same"
> -- hence why I try to use IDEs as little as possible, but I can't knock 'em
> for their advanced features.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Civcraft/PrisonPearl/pull/47/files#r31049577>.
>